src/components/NextStep/NextStep.js

```diff
@@ -7,19 +7,26 @@ import Button from './Button';
 import Wrapper from './Wrapper';
 
 function NextStep({ label = 'Next step', to, onMoveToNextStep, disabled }) {
+  const renderButtonOrLink = () => {
```

Comment:
What do you think about that?
No need to handle errors, no need to duplicate code.

```
function NextStep({ label = 'Next step', onMoveToNextStep = noop, to, disabled }) {
  const renderButtonOrLink = () => {
    const props = {
      ...(to && { as: Link, to }),
      onClick: onMoveToNextStep,
      disabled
    };

    return (
      <Button {...props} data-testid="next-step-button">
        {label}
      </Button>
    );
  };

  return (
    <ErrorBoundary>
      <Wrapper>{renderButtonOrLink()}</Wrapper>
    </ErrorBoundary>
  );
}
```

Reply:
Or something fancy like that
```
const props = {
  ...(to ? { as: Link, to } : { onClick: onMoveToNextStep }),
  disabled
};
```